pkg/health/health.go

```diff
@@ -73,9 +82,15 @@ func (s *Service) Start(context.Context) error {
 	return nil
 }
 
-func (s *Service) Stop(ctx context.Context) error {
+func (s *Service) Stop() error {
 	if s.running.CompareAndSwap(true, false) {
-		return s.srv.Shutdown(ctx)
+		ctx, cancel := context.WithTimeout(context.Background(), s.opts.ShutdownTimeout)
```

Comment:
should this not use the context given by the caller instead of `context.Background()`

Reply:
Well the whole point is to set a timer in the context so it expires. I can use use the same logic outside this package creating the context before calling this function but it seems logically the right place here?

Reply:
I mean, you can wrap the context from the caller instead of creating a new one so i.e. any key that is associated to the parent context would be available here, cancellation of the parent context would trigger the cancellation of children. It may not change much here but I found very useful to wrap contexts instead of creating new ones once a root one is available.

Reply:
solved with https://github.com/sco1237896/sco-backend/pull/15/commits/f1d1b19430d57c43c073a48e5d090d562b7ff628

Reply:
>  It may not change much here but I found very useful to wrap contexts instead of creating new ones once a root one is available.
If anyone can confirm if I got this correctly: wrapping seems to be the way to go in almost every situation since we also want to be notified when/if the parent context is done. Not sure about all the underlying machinery but I'd guess we risk leaking resources if we don't?